core/src/main/java/org/elasticsearch/action/admin/cluster/reroute/TransportClusterRerouteAction.java

```diff
@@ -68,8 +68,18 @@ protected ClusterRerouteResponse newResponse() {
 
     @Override
     protected void masterOperation(final ClusterRerouteRequest request, final ClusterState state, final ActionListener<ClusterRerouteResponse> listener) {
+        ActionListener<ClusterRerouteResponse> logWrapper = ActionListener.wrap(
+            response -> {
+                if (request.dryRun() == false) {
+                    response.getMessages().forEach(logger::info);
+                }
+                listener.onResponse(response);
```

Comment:
Sounds good to me. If that were to happen without `finally`, would an error response with the exception be returned upstream, or would the request handling just time out?